cdap-authorization-dataset-plugin/src/main/java/co/cask/cdap/security/authorization/DatasetBasedAuthorizer.java

```diff
@@ -67,11 +71,22 @@ public ACLDataset get() {
         return new ACLDataset(table);
       }
     };
+    Properties properties = context.getExtensionProperties();
+    if (properties.containsKey("superusers")) {
+      for (String superuser : Splitter.on(",").trimResults().omitEmptyStrings()
+        .split(properties.getProperty("superusers"))) {
+        superUsers.add(new Principal(superuser, Principal.PrincipalType.USER));
+      }
+    }
   }
 
   @Override
   public void enforce(final EntityId entity, final Principal principal,
                       final Action action) throws UnauthorizedException, TransactionFailureException {
+    // no enforcement for superusers
+    if (superUsers.contains(principal)) {
```

Comment:
nit: Is this the right place to include the logic for superUsers? Wondering say if we have a new implementation for AbstractAuthorizer, the developer needs to remember/know that he/she has to handle the superUsers case. Wondering if there is a better way.